### ISSUE TITLE
Pictures don't refresh while PTXprint is running

### ISSUE THREAD

**Issue by davidc86** [sillsdev/ptx2pdf#940]
v2.4.7
If I'm working with pictures and I see a problem, and I fix the picture file.
The updated picture is NOT used when making the PDF.

The only way I have found to refresh the picture inventory is to exit PTXprint and reopen it (not a quick process).

Any clue why PTXprint is not pulling the pictures from the correct location each time it creates a PDF?

**Comment by markpenny** [comment]
My guess is that it is relying on the file in the tmpPics location, and not checking the timestamp on each to see if it needs to be updated. The process of creating the image files in tmpPics is rather complicated, and takes a significant chunk of time each time it needs to run; so we have tried to limit when it does that.

I'll chat with MH to see if there's a relatively quick fix to this while we wait for a more comprehensive overhaul of the system to keep track of dependencies throughout all the temporary files that PTXprint relies on to create the final PDF (only updating anything that really needs to be changed).

**Comment by markpenny** [comment]
@davidc86 Sorry that this has been on hold for so long. MH has been on furlough, so we've doing bare-minimum for a while.

Could I just clarify what you mean by "I see a problem, and I fix the picture file"?

When you say, "fix" the picture file, do you mean you select a different one? Or do you mean something else?

**Comment by davidc86** [comment]
Sorry, I don't recall the exact details that prompted that statement, but here is the issue: if we need to add Selaru text (e.g. city and place names) to a map, and I see in the PDF output that I missed a placename or misspelled something. Or if I see where a picture is showing up with a line along one edge, and I want to erase it (e.g. without using the border erase option). So, then I tweak the picture or fix the map text labels and then rerun PTXprint, but nothing changes! Apparently, PTXprint caches pictures, and if I don't exit and reenter PTXprint, it won't reload the updated pictures in the cache. So, I guess my suggestion was a quick timestamp compare on the cached pictures with the original source pictures just to verify that what PTXprint is going to be using each round is the most up-to-date versions of the pictures. (Or, a simple way to force PTXprint to refresh its cache, which might be just a quick timestamp compare, or just recopying all the pictures; whatever is deemed best. It is just confusing/concerning to users when the picture they just fixed refuses to show up in the output PDF -- especially if they don't understand that PTXprint is using a cache to make things move faster.

Does this help explain that comment?

**Comment by markpenny** [comment]
Yes - that makes sense now, and I can see why it is important for it to always show the latest image. (It didn't make sense earlier as I couldn't see why people would be editing an artists picture; but in the context of a map, it makes perfect sense.)

There is a manual work-around to force the images to be updated:
![image](https://github.com/user-attachments/assets/ffbea3b8-d974-4168-a802-e2628f40b119)

If you _toggle_ the "**Use Other Location**" option on the **Global Options** tab, it will refresh its local cache with the latest version of the image. Not ideal, but quicker than restarting PTXprint! And you should see the updated image right away in the preview window:

![image](https://github.com/user-attachments/assets/7e676984-2601-4879-b6de-d1b2b04713d2)

**Comment by mhosken** [comment]
BTW to save restarting, you can simply change config and change back. It should have the same effect.

**Comment by davidc86** [comment]
Thanks for the tip.
Not obvious to typical user, but good to know it can be forced.
